### PR TITLE
feat: config file

### LINKED ISSUE #4: Config File
Along with using environment variables for configuration, it would be great to have a file for all the config such as
colors, theme,...

For the solution, I suggest an example of kitty terminal config.

image

Additional Context:
Adding an example file to show how to use the config file will make it more clear.

### PR DESCRIPTION
**Pull request by jolheiser**
Resolves #4

Defaults to `$XDG_CONFIG_HOME/nap/config.yaml` or can be set with `NAP_CONFIG`.

Env is umarshaled into the config, then the YAML config is unmarshaled on top of it.
It should only affect values for keys that are set. https://go.dev/play/p/AWdvYaf0JEo

### CODE REVIEW

**Comment by maaslalani** [comment]
Personally I think the ENV should override the config file. What do you think?

It's easier to set ephemeral env vars rather than changing the config file:

```
NAP_HOME=.snippets nap
```

Maybe we read `NAP_CONFIG` manually first and then read the file then the other config.

**Comment by jolheiser** [comment]
You are right, I was thinking about them in reverse. 😅 

I'll send a patch!

**Comment by jolheiser** [comment]
Okay, now the env is parsed _after_ a potential config. 
https://go.dev/play/p/MN8R5F38nx7

File: main.go [hunk 1]
```diff
@@ -112,17 +116,36 @@ func readStdin() string {
...
+		return newConfig()
+	}
+	if fi != nil {
+		defer fi.Close()
+		if err := yaml.NewDecoder(fi).Decode(&config); err != nil {
+			return newConfig()
 		}
 	}
+
+	if err := env.Parse(&config); err != nil {
```
Comment (jolheiser) [inline main.go:145]: I've removed the `envDefault` struct tags, as they were causing this to overwrite config file values even if the env wasn't defined. Now, rather, a config is just pre-created with default values and then overwritten as things unmarshal onto it.

**Comment by maaslalani** [comment]
@jolheiser Looking great! I just added a commit that changes the default config file back to `snippets.json` which I believe is what we need for things to work properly if there's no original file.

**Comment by maaslalani** [comment]
Also gave you collaborator access in case you're interested 🙂

File: main.go [hunk 2]
```diff
@@ -112,17 +116,36 @@ func readStdin() string {
...
+	return cfgPath
+}
+
 // readConfig returns a configuration read from the environment.
 func readConfig() Config {
-	config := Config{Home: defaultHome()}
-	if err := env.Parse(&config); err != nil {
-		return Config{
-			Home:            defaultHome(),
-			File:            defaultSnippetFileName,
```
Comment (jolheiser) [inline main.go:121]: This is where I grabbed the default snippet filename from, but after review I think you're correct it should be `snippets.json`

Comment (maaslalani) [reply main.go:121]: Yeah sorry about the confusion, I think that was incorrect originially but the `envDefault` was overriding it.

**Pull request merged**